core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -234,11 +242,45 @@ void processSubchannelState(Subchannel subchannel, ConnectivityStateInfo stateIn
     }
   }
 
+  private void updateHealthCheckedState(SubchannelData subchannelData) {
+    if (subchannelData.state != READY || (concludedState == TRANSIENT_FAILURE
```

Comment:
if `concludedState == TRANSIENT_FAILURE && healthState != READY`, then we don't need to change the state, but we may need to update the picker with a newer error message.

Reply:
Sounds fair.